server/jetstream_events.go

```diff
@@ -110,6 +110,21 @@ type JSConsumerDeliveryExceededAdvisory struct {
 // JSConsumerDeliveryExceededAdvisoryType is the schema type for JSConsumerDeliveryExceededAdvisory
 const JSConsumerDeliveryExceededAdvisoryType = "io.nats.jetstream.advisory.v1.max_deliver"
 
+// JSConsumerDeliveryNakAdvisory is an advisory informing that a message was
+// naked by the consumer
+type JSConsumerDeliveryNakAdvisory struct {
+	TypedEvent
+	Stream      string `json:"stream"`
+	Consumer    string `json:"consumer"`
+	ConsumerSeq uint64 `json:"consumer_seq"`
+	StreamSeq   uint64 `json:"stream_seq"`
+	Deliveries  uint64 `json:"deliveries"`
+	Domain      string `json:"domain,omitempty"`
```

Comment:
We dont have domain on the others, probably something we should consider doing, good call

Reply:
Should we do that here, adding domain to the others or in separate PR?

Reply:
I've removed the draft, so you @ripienaar can merge this PR. I am not quite following in which other places the domain should be added. As far as I see, all Jetstream advisories do have it.

Reply:
You're right, sorry for the distration, I missed that they were added previously.  So I think we're good.